htmltest/check-link.go

```diff
@@ -347,6 +348,70 @@ func (hT *HTMLTest) checkMailto(ref *htmldoc.Reference) {
 		})
 		return
 	}
+
+	// split off domain, check mx, fallback to A or AAAA if that fails
+	var dnserr *net.DNSError
+	var ok bool
+
+	domain := strings.Split(ref.URL.Opaque, "@")[1]
+
+	for domain != "" {
+		// if a simple MX lookup works, we are done, continue
+		if _, err := net.LookupMX(domain); err == nil {
+			break // success, time to exit
+		} else if dnserr, ok = err.(*net.DNSError); !ok || dnserr.Err != "no such host" {
+			// this isn't an error we are expecting to see here
+			hT.issueStore.AddIssue(issues.Issue{
+				Level:     issues.LevelWarning,
+				Message:   "unable to perform LookupMX, unknown error",
+				Reference: ref,
+			})
+			return
+		}
+
+		// do we have to restart because of a CNAME
+		if cname, err := net.LookupCNAME(domain); err == nil && cname != domain {
+			// we have a valid CNAME, try with that. Loops return NXDOMAIN by default
+			domain = cname
+			continue
+
+		} else if dnserr, ok = err.(*net.DNSError); !ok || dnserr.Err != "no such host" {
+			// this isn't an error we are expecting to see here
```

Comment:
The exact error on Linux for the loop CNAME test is `server misbehaving` (value of `dnserr.Err`) which seems in-line-ish with nslookup:

```
~ ❯ nslookup loop1.dtaylor.uk
Server:		127.0.1.1
Address:	127.0.1.1#53

** server can't find loop1.dtaylor.uk: SERVFAIL
```

Likely an error discrepancy between Linux and OSX. You may need to restructure a bit to handle this cleanly.

Reply:
I'm not sure if this is infact an issue, no such host is an isolated case where we want to proceed to checking A records. Should a CNAME loop not produce an warning of some variety?
I accept that this may be inconsistent between platforms, but I feel the Linux way is actually better in this case and the OS X way is an acceptable fallback?